galleon-plugins/src/main/java/org/wildfly/galleon/plugin/server/ForkedEmbeddedUtil.java

```diff
@@ -96,7 +98,13 @@ public static int fork(ForkCallback callback, Path props, String... args) throws
                     e.printStackTrace();
                 }
             }
-            return p.exitValue();
+
+            int exitCode = p.exitValue();
+            if (exitCode == FAILED){
```

Comment:
Let's use if (exitCode != 0) instead. And let's modify the method to not return an int but void. This exit code outside this method doesn't seem to be useful for anything.